Weekly_Work/Week2/Metrics_Root_class.py

```diff
@@ -0,0 +1,98 @@
+import json
+import datetime
+import matplotlib.pyplot as plt
+import pandas as pd
+
+
+class Metric():
+    
+    def __init__(self, path, source_code_exclude_list=None):
+
+        self.source_code_exclude_list = source_code_exclude_list
+        self.repo_urls = set()
```

Comment:
Why do you need this?

Reply:
I have removed the source_code_list variable to the commit class since source code can actually be measured only for commits and put requests.
The repo_urls is used since even though data category may be homogenous, it can have info from several repositories and thus, this variable allows for creating a dictionary which has the metric value for each repo.

Reply:
OK, I see. But i see two aspects here:

* First, we have the repo name as a part of each item, so that we could use from there, if needed.
* Second, up to now, metrics have been defined in the context of a single repository. However, for having it computed on a set of repositories (which is interesting) it is good enough to have a dataset when instantiating the class with all the involved repositories.

All in all, I suggest that if we want to compute a given metric for each of a given set of repos, we just filter the items before creating the metrics object, and we just create a metrics object per repo. That way, the metrics object (class) will be a bit less complex, which in general is a good thing to have.